### DISCUSSION TITLE
extract 12 alpha phase variants in a selected beta grain

### DISCUSSION THREAD

**Discussion by Lu-Peter** [mtex-toolbox/mtex#2013] [Ask Anything] [answered]
I used the following code to extract the 12 12 alpha phase variants in a selected beta grain, but the area fractions alway return zero. Could you please help to correct the code? Thank you very much.

% to extract 12 alpha phase variants in a selected beta grain
csBeta =  CS{3}; % beta phase
beta2alpha = orientation.Burgers(csBeta, ebsd('Ti-Hex').CS);
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'), 'threshold', 3*degree);
job = parentGrainReconstructor(ebsd, grains);

job.p2c = beta2alpha;
job.calcVariantGraph('threshold', 3*degree);
job.clusterGraph('numIter', 3);
job.calcParentFromVote;
job.mergeSimilar('threshold', 5*degree);
job.mergeInclusions('maxSize', 50);
parentEBSD = job.ebsd;
[parentGrains, parentEBSD.grainId, parentEBSD.mis2mean] = calcGrains(parentEBSD, 'angle', 10*degree);

parentGrains = smooth(job.parentGrains, 10);

% Plot the mean orientation of parent grains
figure;
plot(parentGrains, parentGrains.meanOrientation)

% Get user input for selecting a grain boundary
[x, y] = ginput(1);
grainSelected = parentGrains(parentGrains.findByLocation([x, y]));

% Highlight the selected grain boundary
hold on
plot(grainSelected.boundary, 'linewidth', 3, 'lineColor', 'w');
hold off

% Further analysis on child grains
childGrains = job.grainsPrior(job.mergeId == grainSelected.id);
childGrains = childGrains(ebsd('Ti-Hex').CS);

% Plot child grain orientations
figure;
childOri = job.ebsdPrior(childGrains).orientations;
parentOri = grainSelected.meanOrientation;
[variantId, packetId] = calcVariantId(parentOri, childOri, job.p2c);

tol=varEBSD.id(1:length(varEBSD));
v=[];
symnum = [1, 2, 3, 7, 8, 9, 10, 11, 12, 13, 14, 15];
% Plot the child grains for each variant
for k = 1:12
    varEBSD = ebsd(childGrains);
    figure;
   tol=varEBSD.id(1:length(varEBSD)); % varEBSD's id does not start from 1
    for j = 1:length(varEBSD)
        if varEBSD('id', tol(j)).phaseId == 3  %3 for beta phase
            ori1 = varEBSD('id', tol(j)).orientations;
            temori = grainSelected.meanOrientation;
            temorisym = temori.symmetrise;
	ori2 = temorisym(symnum(k))*inv(beta2alpha);
	if angle(ori1, ori2)./degree > 5
	varEBSD('id', tol(j)) = [];
	end
	else
	varEBSD('id', tol(j)) = [];
	end
	end
v(k) = length(varEBSD)
figure;
plot(varEBSD, varEBSD.orientations);
end

**Comment by AzdiarGazder** [comment, marked as answer]
HI @Lu-Peter,

Please have a look at [ORTools](https://github.com/ORTools4MTEX/ORTools).
It comprises a ready set of scripts that help you undertake detailed analysis of your variants.

Warm regards,
Azdi

**Comment by Lu-Peter** [comment]
Thanks Azdi. Frank mentioned this to to me. I forgot this. I tried it. Very useful ! Thanks so much.
